### PR TITLE
Update the Slack import docs with latest features.

### PR DESCRIPTION
**Pull request by grundleborg**
This updates the Slack import docs, in particular to note the fact that
Bot messages, and file attachments can now be imported.

Please let me know what you think of this, as I'm sure there's room to improve the quality of expression and clarity of information.

https://mattermost.atlassian.net/browse/PLT-4280

### CODE REVIEW

File: source/administration/migrating.rst [hunk 1]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
 Migrating from Slack
 ====================
 
-You can import channels and users from Slack into Mattermost from Team Settings > Import. Please review our documentation on `Slack Import <https://docs.mattermost.com/help/settings/team-settings.html#import-from-slack-beta>`_ for more details.
+.. note:: As a proprietary SaaS service, Slack is able to change its export format quickly and without notice. If you encounter issues not mentioned in the documentation below, please alert the product team by `filing an issue <https://github.com/mattermost/platform/issues>`_.
```
Comment (yangchen1) [inline source/administration/migrating.rst]: "Note:" should be in bold and the following text should be on the same line

File: source/administration/migrating.rst [hunk 2]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+The Slack Import feature in Mattermost is in "Beta" and focus is on supporting migration of teams of less than 100 registered users.
```
Comment (yangchen1) [inline source/administration/migrating.rst]: "and focus is on" -> "and focused on"

File: source/administration/migrating.rst [hunk 3]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+.. warning:: **It is highly recommended that you test Slack import before applying it to an instance intended for production.**
+
+   If you use Docker, you can spin up a test instance in one line (``docker run --name mattermost-dev -d --publish 8065:80 mattermost/platform``). If you don't use Docker, there are `step-by-step instructions <http://docs.mattermost.com/install/docker-local-machine.html>`_ to install Mattermost in preview mode in less than 5 minutes.
+
+Supported Features
+++++++++++++++++++
+
+The following key features are supported out of the box when importing from Slack:
+
+* User accounts with and email address set.
```
Comment (yangchen1) [inline source/administration/migrating.rst]: with an*

File: source/administration/migrating.rst [hunk 4]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
 ====================
 
-You can import channels and users from Slack into Mattermost from Team Settings > Import. Please review our documentation on `Slack Import <https://docs.mattermost.com/help/settings/team-settings.html#import-from-slack-beta>`_ for more details.
+.. note:: As a proprietary SaaS service, Slack is able to change its export format quickly and without notice. If you encounter issues not mentioned in the documentation below, please alert the product team by `filing an issue <https://github.com/mattermost/platform/issues>`_.
+
+The Slack Import feature in Mattermost is in "Beta" and focus is on supporting migration of teams of less than 100 registered users.
+
+This feature can be accessed through the `Mattermost Web App </administration/migrating.html#migrating-from-slack-using-the-mattermost-web-app>`_ or using the `CLI </administration/migrating.html#migrating-from-slack-using-the-mattermost-cli>`_.
+
+.. warning:: **It is highly recommended that you test Slack import before applying it to an instance intended for production.**
```
Comment (yangchen1) [inline source/administration/migrating.rst]: "Warning:" should be in bold and the following text should be on the same line

File: source/administration/migrating.rst [hunk 5]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+* Channel topic and purpose.
+
+* Automatically adding imported users to their channels.
+
+Messages with file attachments are imported as a message containing a link to Slack's servers by default. The file attachments themselves can be imported to Mattermost by using the `Slack Advanced Exporter <https://github.com/grundleborg/slack-advanced-exporter>`_ tool to add them to your archive before importing it.
+
+Bot and Integration messages are imported by default, but if you would like them to display with the appropriate username when imported, you should ensure that `Enable Integrations to Override Usernames </administration/config-settings.html#enable-integrations-to-override-usernames>`_ is set **before** doing the import.
+
+.. note:: Slack user accounts with the same email address as existing accounts on your Mattermost server will be merged into those accounts on import.
```
Comment (yangchen1) [inline source/administration/migrating.rst]: "Note:" should be in bold and the following text should be on the same line

File: source/administration/migrating.rst [hunk 6]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+   If you use Docker, you can spin up a test instance in one line (``docker run --name mattermost-dev -d --publish 8065:80 mattermost/platform``). If you don't use Docker, there are `step-by-step instructions <http://docs.mattermost.com/install/docker-local-machine.html>`_ to install Mattermost in preview mode in less than 5 minutes.
+
+Supported Features
+++++++++++++++++++
+
+The following key features are supported out of the box when importing from Slack:
+
+* User accounts with and email address set.
+
+* Public channels and text messages posted in them by users.
```
Comment (yangchen1) [inline source/administration/migrating.rst]: and the***, makes it a bit easier to read

File: source/administration/migrating.rst [hunk 7]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+The following limitations are present when importing from Slack:
+
+* The Markdown subset used by Slack's Posts 2.0 feature announced on September 28, 2015 is not yet supported.
+
+* Direct Message and Private Group history. Slack does not include these messages when generating the export archive.
+
+Migrating from Slack using the Mattermost Web App
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+.. note:: For larger imports, particularly those where you have used the `slack-advanced-exporter tool` to add Slack post attachments to the archive, it is recommended to import the Slack data using the `CLI </administration/migrating.html#migrating-from-slack-using-the-mattermost-cli>`_.
```
Comment (yangchen1) [inline source/administration/migrating.rst]: "Note:" should be in bold and the following text should be on the same line

File: source/administration/migrating.rst [hunk 8]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+.. note:: Slack user accounts with the same email address as existing accounts on your Mattermost server will be merged into those accounts on import.
+
+Limitations
++++++++++++
+
+The following limitations are present when importing from Slack:
+
+* The Markdown subset used by Slack's Posts 2.0 feature announced on September 28, 2015 is not yet supported.
+
+* Direct Message and Private Group history. Slack does not include these messages when generating the export archive.
```
Comment (yangchen1) [inline source/administration/migrating.rst]: Direct Messages and Private Groups cannot be imported.

File: source/administration/migrating.rst (hunk 1 above)
Comment (grundleborg) [inline source/administration/migrating.rst]: Not changed, as discussed on pre-release.

File: source/administration/migrating.rst (hunk 5 above)
Comment (grundleborg) [inline source/administration/migrating.rst]: Not changed, as discussed on pre-release.

File: source/administration/migrating.rst (hunk 7 above)
Comment (grundleborg) [inline source/administration/migrating.rst]: Not changed, as discussed on pre-release.

File: source/administration/migrating.rst [hunk 9]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
 Migrating from Slack
 ====================
 
-You can import channels and users from Slack into Mattermost from Team Settings > Import. Please review our documentation on `Slack Import <https://docs.mattermost.com/help/settings/team-settings.html#import-from-slack-beta>`_ for more details.
+.. note:: As a proprietary SaaS service, Slack is able to change its export format quickly and without notice. If you encounter issues not mentioned in the documentation below, please alert the product team by `filing an issue <https://github.com/mattermost/platform/issues>`_.
+
+The Slack Import feature in Mattermost is in "Beta" and focused on supporting migration of teams of less than 100 registered users.
+
+This feature can be accessed through the `Mattermost Web App </administration/migrating.html#migrating-from-slack-using-the-mattermost-web-app>`_ or using the `CLI </administration/migrating.html#migrating-from-slack-using-the-mattermost-cli>`_.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: Can we reference to the links without having `https://docs.mattermost.com` parts first?

File: source/administration/migrating.rst [hunk 10]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+.. warning:: **It is highly recommended that you test Slack import before applying it to an instance intended for production.**
+
+   If you use Docker, you can spin up a test instance in one line (``docker run --name mattermost-dev -d --publish 8065:80 mattermost/platform``). If you don't use Docker, there are `step-by-step instructions <http://docs.mattermost.com/install/docker-local-machine.html>`_ to install Mattermost in preview mode in less than 5 minutes.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: Propose

```
If you use Docker, you can spin up a test instance in one line:

   .. code:: bash

       docker run --name mattermost-dev -d --publish 8065:80 mattermost/platform

If you don't use Docker, there are `step-by-step instructions <http://docs.mattermost.com/install/docker-local-machine.html>`_ to install Mattermost in preview mode in less than 5 minutes.
```

for consistency with other docs, e.g. http://docs.mattermost.com/install/docker-local-machine.html

File: source/administration/migrating.rst [hunk 11]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+Supported Features
+++++++++++++++++++
+
+The following key features are supported out of the box when importing from Slack:
```
Comment (jasonblais) [inline source/administration/migrating.rst]: `out of the box` seems unnecessary

File: source/administration/migrating.rst [hunk 12]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+* User accounts with an email address set.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: no periods for this bullet list as the items aren't complete sentences

File: source/administration/migrating.rst [hunk 13]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+* Public channels and the text messages posted in them by users.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: is the `by users` part intentional? E.g. are bot messages not imported?

File: source/administration/migrating.rst [hunk 14]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+* Channel topic and purpose.
+
+* Automatically adding imported users to their channels.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: Reword to make the phrasing more consistent with previous bullet points, which start with a noun-adjective pair, e.g. `public channels`, `channel topic`

Perhaps `Imported users added automatically to their channels`

Only 1/5 here though,

File: source/administration/migrating.rst [hunk 15]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+Messages with file attachments are imported as a message containing a link to Slack's servers by default. The file attachments themselves can be imported to Mattermost by using the `Slack Advanced Exporter <https://github.com/grundleborg/slack-advanced-exporter>`_ tool to add them to your archive before importing it.
+
+Bot and Integration messages are imported by default, but if you would like them to display with the appropriate username when imported, you should ensure that `Enable Integrations to Override Usernames </administration/config-settings.html#enable-integrations-to-override-usernames>`_ is set **before** doing the import.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: Same thing with the link, not sure if `</administration...>` is enough? (might be, just don't know)

Maybe add `you should ensure that 'Enable Integrations .... is set in **System Console > Integrations > Custom Integrations** ....`

File: source/administration/migrating.rst [hunk 16]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+.. note:: Slack user accounts with the same email address as existing accounts on your Mattermost server will be merged into those accounts on import.
+
+Limitations
++++++++++++
+
+The following limitations are present when importing from Slack:
+
+* The Markdown subset used by Slack's Posts 2.0 feature announced on September 28, 2015 is not yet supported.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: Non-documentation question: Wondering if we could create a Help Wanted issue for it?

Someone might be interested in adding support for it.

File: source/administration/migrating.rst [hunk 17]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+* Direct Message and Private Groups cannot be imported. Slack does not include these messages when generating the export archive.
+
+Migrating from Slack using the Mattermost Web App
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+.. note:: For larger imports, particularly those where you have used the `slack-advanced-exporter tool` to add Slack post attachments to the archive, it is recommended to import the Slack data using the `CLI </administration/migrating.html#migrating-from-slack-using-the-mattermost-cli>`_.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: same note on URL as above

File: source/administration/migrating.rst [hunk 18]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+
+1. Generate a Slack "Export" file from **Slack > Team Settings > Import/Export Data > Export > Start Export**.
+
+2. In Mattermost go to **Team Settings > Import > Import from Slack**. *Team Administrator* or *System Administrator* role is required to access this menu option.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: I don't think we usually italicize Team Admin and System Admin.

We also use "Team Admin" and "System Admin" when defining the roles in our docs: https://docs.mattermost.com/help/getting-started/managing-members.html#user-roles

File: source/administration/migrating.rst [hunk 19]
```diff
@@ -26,4 +26,70 @@ The following instructions migrate Mattermost from one server to another by back
...
+   ``$ platform -slack_import -team_name="your-team" -import_archive /path/to/your-slack-export.zip``
+
+Using the Imported Team
++++++++++++++++++++++++
+
+* During the import process, the emails and usernames from Slack are used to create new Mattermost accounts.
+
+* Slack users can activate their new Mattermost accounts by using Mattermost's Password Reset screen with their email addresses from Slack to set new passwords for their Mattermost accounts.
+
+* Once logged in, the Mattermost users will have access to previous Slack messages in the public channels imported from Slack.
```
Comment (jasonblais) [inline source/administration/migrating.rst]: remove `the` in `the Mattermost users`

**Comment by grundleborg** [comment]
@jasonblais and @yangchen1 - Thanks for the reviews, I've now addressed all your comments.

**Comment by jasonblais** [comment]
Really minor formatting tweak that I missed on first read, which I committed to the branch.

Otherwise looks good, great work here!

**Pull request merged**